src/legacy/core_plugins/metrics/public/np/plugin.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { tsvb } from '../tsvb_fn';
+// @ts-ignore
+import { MetricsVisProvider } from '../kbn_vis_types';
+import { TypesService } from '../../../visualizations/public/types/types_service';
+import { InterpreterService } from '../interpreter';
+
+export class Plugin {
+  private readonly visTypes: TypesService;
+  private readonly interpreter: InterpreterService;
+
+  constructor() {
+    this.visTypes = new TypesService();
+    this.interpreter = new InterpreterService();
+  }
+
+  public setup() {
+    this.interpreter.setup().functionsRegistry.register(tsvb);
+    // register the provider with the visTypes registry so that other know it exists
+    this.visTypes.setup().VisTypesRegistryProvider.register(MetricsVisProvider);
```

Comment:
Similar to my note on the interpreter dependency, it would be best to pass `visualizations` as part of the second argument to `setup` as this will mirror the way it works in the new platform:

```javascript
// probably put this import in your index/shim file
import { visualizations } from '../../../visualizations';

// ...
setup(core, plugins) {
  plugins.visualizations.visTypes.VisTypesRegistryProvider.register(MetricsVisProvider);
}
```

As a rule of thumb, there should never be a need to import something beyond the top-level of another plugin, as paths within a plugin are considered implementation details in the new platform and are not part of the contract.

Reply:
Fixed. Thanks